content/2022-11-17-drogue-cloud-zero-eleven/index.md

```diff
@@ -0,0 +1,130 @@
+it makes sense to provide them out of the box. One might call that "IoT middleware", or "Digital Twin", as it
+works with the mirrored state of the devices. The name of the project is
+[Doppelgänger](https://en.wikipedia.org/wiki/Doppelg%C3%A4nger), and it mainly provides digital twin functionality.
+
+The goals of Doppelgänger are:
+
+* Work with any IoT connectivity layer: True, we focus on Drogue Cloud. But with a bit of code, it should easily be possible to attach it to e.g. a standard MQTT broker.
+* Normalize IoT payload: The tricky part. As you never know what data you get, unless you adopt some existing schema. And this is what we aim for. We do have a simple internal data model, which should be able to support existing data models, like OPC UA. And we like with the connectivity layer, we provide a translation layer to convert data into the Doppelgänger thing model.
+* Provide out of the box functionality to work with data: Renaming properties, transforming values, evaluating state, reconciling between desired and actual state, aggregating data, building thing relations.
+* Provide APIs to the things state and configuration: Not only our own APIs, but also support APIs like OPC UA or Web of Things.
+
+So, if these are the goals, what is the current state of the project? We do have a 0.1 release. That sounds early, and it is. But it is the right time to check it out, and come up with your feedback 😉.
+
+And just to make you a bit curious, here are a few features the 0.1 release already bring:
+
+* IoT connectivity with Drogue Cloud
+* Updating state from JSON messages
+* State reconciliation with custom JavaScript code (based on state change, timers, or lifecycle events)
+* State aggregation and a dynamic hierarchy using references
+* Validation of the things state using a JSON Schema
+* Reconciliation of desired state based on actual state (e.g. drive the actual state towards the desired state using commands, with re-try and scripts)
+* Subscribing to the thing state using WebSocket and OPC UA
+
+We do have a set of [example snippets](https://github.com/drogue-iot/drogue-doppelgaenger/tree/main/examples), which might show this in more technical way.
+
+<figure>
+<img src="twin1.png" alt="Screenshot #1">
+<figcaption>Root of the hierarchy in the debugger web application</figcaption>
+</figure>
+
+<figure>
+<img src="twin2.png" alt="Screenshot #2">
+<figcaption>A thing in the web debugger</figcaption>
+</figure>
+
+<figure>
+<img src="twin3.png" alt="Screenshot #3">
+<figcaption>The same hierarchy in an OPC UA client</figcaption>
+</figure>
+
+And to give you a bit of context to the screenshots. This shows the web based debugger on a Doppelgänger instance we used for IoT hackathon at EclipseCon 2022. The data came from micro:bits, through a Bluetooth Mesh network, connected via gateways to Drogue Cloud. The OPC UA twin integration shows the same data, just using OPC UA as the API instead of a simple WebSocket API.
```

Comment:
```suggestion
And to give you a bit of context to the screenshots, this shows the web based debugger on a Doppelgänger instance we used for the IoT hackathon at EclipseCon 2022. The data came from micro:bits, through a Bluetooth Mesh network, connected via gateways to Drogue Cloud. The OPC UA twin integration shows the same data, just using OPC UA as the API instead of a simple WebSocket API.
```